### PR TITLE
Setup "webpack encore" with dependence for Vue Js

### PR DESCRIPTION
**Pull request by senzowayne**

### CODE REVIEW

**Comment by yalefresne** [comment]
I request @Rantignac review cause he shared a config for webpack very interesting with me, but i don't refound  it.  
So @Rantignac, if you can share it again here ^^

**Comment by Rantignac** [comment]
@yalefresne, @senzowayne, we can use the webpack encore aliases  to have namespaces in JS: https://blog.elao.com/fr/dev/webpack-encore-alias-namespace/

**Comment by senzowayne** [comment]
Amazing !! 🔥 thanks for share.

**Comment by yalefresne** [comment]
Yeah ! Thanks for sharing @Rantignac !  
So @senzowayne if you could set that, that will be nice ! 🚀
Globally, if you could delete all not necessary comments, but of course, you can leave that you think use later.

**Comment by senzowayne** [comment]
I noticed an improvement in speed during the build (with namespace) 🔥

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,10 @@
 /docker/mysql
 ### docker ###
 
+###> idea ###
+/.idea
+### idea ###
+
```
Comment (yalefresne) [inline .gitignore]:
```suggestion
###> idea ###
/.idea
### idea ###

```
You should put this lines in a personnal gitignore file.  
[https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file](https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file)

File: assets/app.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import './styles/app.css';
+
+console.log('Hello Webpack Encore! Edit me in assets/app.js');
```
Comment (yalefresne) [inline assets/app.js:3]: I'm not sure, we need this line ^^

**Review by yalefresne — CHANGES_REQUESTED** [review]

File: assets/app.js (hunk 2 above)
Comment (yalefresne) [inline assets/app.js]:
```suggestion
console.log('Hello Webpack Encore! Edit me in assets/app.js');
```
I'm not sure we need this line.

**Review by yalefresne — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+    "devDependencies": {
+        "@symfony/webpack-encore": "^0.31.0",
```
Comment (yalefresne) [inline package.json:4]:
```suggestion
        "@babel/core": "^7.12.3",
        "@symfony/webpack-encore": "^0.31.0",
```
To resolve some dependency warnings after `yarn install`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "@vue/babel-helper-vue-jsx-merge-props": "^1.2.1",
+        "@vue/babel-preset-jsx": "^1.2.2",
+        "core-js": "^3.0.0",
```
Comment (yalefresne) [inline package.json:7]:
```suggestion
        "core-js": "^3.0.0",
        "css-loader": "^5.0.0",
```
To resolve `warning " > vue-loader@15.9.3" has unmet peer dependency "css-loader@*".`

**Review by yalefresne — CHANGES_REQUESTED** [review]

**Review by yalefresne — APPROVED** [review]

**Pull request merged**